src/main/java/dk/kb/image/api/v1/impl/AccessApiServiceImpl.java

```diff
@@ -373,6 +408,121 @@ public StreamingOutput iIPImageRequest(
         }
     }
 
+    
+    protected StreamingOutput createStreamingOutput(final byte[] data) {
+        return new StreamingOutput() {
+            @Override
+            public void write(OutputStream output) throws IOException {
+                output.write(data);
+            }
+        };
+    }
+
+    private StreamingOutput getImageForbidden() throws Exception {
+        BufferedImage image = BufferedImageUtils.getNoAccessImage();
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ImageIO.write(image, "jpg", baos);
+        byte[] bytes = baos.toByteArray();
+
+        return createStreamingOutput(bytes);
+
+    }
+
+    private StreamingOutput getImageNotExist() throws Exception {
+        BufferedImage image = BufferedImageUtils.getImageNotExist();
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ImageIO.write(image, "jpg", baos);
+        byte[] bytes = baos.toByteArray();
+        return createStreamingOutput(bytes);
+    }
+    
+    private StreamingOutput handleNoAccessOrNoImage(String resource_id)  throws Exception{
+        ACCESS_TYPE type = accessTypeToImage( resource_id);
+
+        log.debug("Access type:" + type + " for resource_id:" +  resource_id);
+
+        switch (type) {
+        case NO_ACCESS:
+            // Maybe http 403 (forbidden) also?
+            return getImageForbidden();
+        case ID_NON_EXISTING:
+            //Maybe  404  also?
```

Comment:
We like 404 and I'm pretty sure we will put in our own image and handling of 404 in this case. It is correct that the onerror handler on img tag  will not fire if an image is supplied and this seems to be according to spec as the onerror handler is related to disturbance of the img load process_ and not the http code. So if the 404 is accompanied by an image place holder the onerror will not trigger.

It does not make any difference for us because we catch and handle the error elsewhere. 

Is there something in the design that maybe makes it relevant for us to serve the 404 img as is?

Reply:
@jorntx   has spoken. 
http 403 and 404 Implemented for the two cases.